Question_9.py

```diff
@@ -0,0 +1,94 @@
+"""
+9. Take two datetimes from user start_time and end_time. 
+Calculates the difference between start_time and end_time by removing night intervals (12 AM to 6 AM). 
+(Hint: You can use python intervals module)         
+2023-08-30 7:09:59.999999
+"""
+from datetime import datetime, time, timedelta
+
+start_time = datetime.strptime(
+    input(
+        "Enter Start time in format of YYYY-MM-DD hh:mm:ss.milisecs : ",
+    ),
+    "%Y-%m-%d %H:%M:%S.%f",
+)
+end_time = datetime.strptime(
+    input(
+        "Enter End time in format of YYYY-MM-DD hh:mm:ss.milisecs : ",
+    ),
+    "%Y-%m-%d %H:%M:%S.%f",
+)
+
+t2 = timedelta(
+    hours=int(start_time.time().hour),
+    minutes=int(start_time.time().minute),
+    seconds=int(start_time.time().second),
+    microseconds=int(start_time.time().microsecond),
+)
+
+t3 = timedelta(
+    hours=int(end_time.time().hour),
+    minutes=int(end_time.time().minute),
+    seconds=int(end_time.time().second),
+    microseconds=int(end_time.time().microsecond),
+)
+
+
+diff = end_time - start_time  # timedelta object
+
+
+
+if diff.days < 1:
+    
+    if t2 > timedelta(hours=6, minutes=0, seconds=0, microseconds=0):
+        day_first = timedelta(hours=23, minutes=59, seconds=59, microseconds=999999) - t2 
+    else:
+        day_first = timedelta(hours=0, minutes=0, seconds=0, microseconds=0)
+
+    print('day_first',day_first)
+    
+    if t3 > timedelta(hours=6, minutes=0, seconds=0, microseconds=0):
+        day_last = t3 - timedelta(hours=6, minutes=0, seconds=0, microseconds=0)
+    else:
+        day_last = timedelta(hours=0, minutes=0, seconds=0, microseconds=0)
+
+    print('day_last',day_last)        
+
+    # if the difference is less than a day then we need to check for hour and minute differences as well
+    x = timedelta(days=(diff.days )).total_seconds()/3600
+    print('x',x)
+    y = timedelta(hours=x * 0.75)
```

Comment:
For what purpose you are multiplying with 0.75?

Reply:
> For what purpose you are multiplying with 0.75?

sir instead of subtracting a night interval of 12 am to 6 am from every single day and then adding it to the aggregate sum of the total time, I am simply multiplying the number of days in between intervals with 0.75.

here 6 hrs of 24 hrs in a day which is 25 %  of a day or 0.25 which we need to subtract from a single day which means the time we want is 0.75 or 75 % of a day.

Reply:
> But are you sure that you can calculate exact percentage? Let's say if percentage is like 33.333333.... then can you consider this as exact percentage?
> 
> Let me know if you have any solution that can be consider as robust solution.

okay, sir let me find a robust solution.

Reply:
Check python `Portion` module.

Reply:
> But are you sure that you can calculate exact percentage? Let's say if percentage is like 33.333333.... then can you consider this as exact percentage?
> 
> Let me know if you have any solution that can be consider as robust solution.

Sir, I have optimized my considering your concerns. Now it is dynamic with night time whether it is 6 or 7 or 7:05

Reply:
But still the logic is bit complicated. Try to rewrite logic using Portion module. It will reduce your code as well as make it more readable.

Reply:
> But still the logic is bit complicated. Try to rewrite logic using Portion module. It will reduce your code as well as make it more readable.

Okay sir, that may be the case but I was thinking of time complexity as it is O(1) of my solution. But yeah you are right that the readability of code is also important. 
Thank you for your valuable feedback sir.